pkgs/development/interpreters/elixir/generic-builder.nix

```diff
@@ -70,82 +73,84 @@ let
 
   erlc_opts = [ "deterministic" ] ++ optionals debugInfo [ "debug_info" ];
 in
-assert assertMsg (versionAtLeast (getVersion erlang) minimumOTPVersion) compatibilityMsg;
-assert assertMsg maxAssert compatibilityMsg;
-
-stdenv.mkDerivation {
-  pname = "${baseName}";
-
-  inherit src version debugInfo;
-
-  nativeBuildInputs = [ makeWrapper ];
-  buildInputs = [ erlang ];
-
-  env = {
-    LANG = "C.UTF-8";
-    LC_TYPE = "C.UTF-8";
-    DESTDIR = placeholder "out";
-    PREFIX = "/";
-    ERL_COMPILER_OPTIONS = "[${concatStringsSep "," erlc_opts}]";
-  };
-
-  preBuild = ''
-    patchShebangs ${escriptPath} || true
-  '';
-
-  # copy stdlib source files for LSP access
-  postInstall = ''
-    for d in lib/*; do
-      cp -R "$d/lib" "$out/lib/elixir/$d"
-    done
-  '';
+if !config.allowAliases && !bothAssert then
+  # Don't throw without aliases to not break CI.
+  null
+else
+  assert assertMsg bothAssert compatibilityMsg;
```

Comment:
Based on the description of the PR, I think you just need to switch between fail-fast and fail-safe. Correct me if I'm wrong! So, can we define an `assert` function that can switch between the fail-fast and fail-safe to make sure the condition of `if` is exactly the same as the condition of assertion? It will also help you avoid redundant if-else all over this PR.

Reply:
they are not the same! that's my point! why are they different?

your other PR today makes more sense, since you moved the assertion so they will be run lazily, but the changes here feels wrong!

Reply:
Thank you for explanation. The `assertIf` is exactly what I was referring to, but in the end it makes sense. Even though it would be better if we had a trace of this deferring implementation in code directly by commenting a TODO, but I approved this PR.